machAeroTutorials/overset_tips.rst

```diff
@@ -0,0 +1,134 @@
+
+.. figure:: images/overset_guide_15.jpg
+    :align: center
+    :width: 400
+
+    ADflow output with a successful hole cutting. This results in a valid overset mesh.
+..
+    src: overset_guide.pdf page 15
+
+The following points indicate a problem:
+
+* Several flooding iterations
+* Small number of compute cells
+* Orphan cells are present
+
+.. figure:: images/overset_bad_IHC.png
+    :align: center
+    :width: 400
+
+    Bad IHC output.
+
+Flood troubleshooting
+---------------------
+
+If the mesh is flooding (too many flooding iterations, a high number of flooded cells), we need to first prevent this to get a valid hole cutting.
+For this, we need to check leaks in the flooding process:
+
+#. Set the ADflow option: ``"nRefine": 1``.
+   This stops the IHC algorithm after one iteration.
+   You will get a warning, but this is fine.
+   We just want to get an intermediate output for visualization.
+   You can also modify the ``nFloodIter`` option to control how many *flooding* iterations are performed.
+   For example, if ADflow segfaults in the first overset iteration because the whole mesh floods, then you can stop the flooding iterations early by setting ``nFloodIter`` to 1 or 2.
+   A value of 1 will just determine the flood seeds, a value of 2 will do a first pass of the flooding process.
+#. Set ADflow option: ``"useZipperMesh": False``.
+   This skips the zipper mesh generation, which may crash if the hole cutting does not work.
+#. Run the overset check file: ``ihc_check.py``.
+   (This can be found under the tutorial/overset/mesh directory in this repo.)
+#. Open the output volume file in Tecplot.
+#. Use the blanking option to show only ``iBlank`` = -3 (flood seeds) and ``iBlank`` = -2 (flooded cells).
+#. Check which CGNS blocks are fully flooded.
+   Identify the flood cells that connect cells inside the body to cells outside the body.
+   This is where the leak occurs.
+
+The following points might help to fix your flooding issue.
+Check them first.
+
+Check if the meshes have similar growth ratios in the pyHyp extrusion.
+    Flooding is usually caused by cells that grow too fast off a wall.
+    A mesh with a high growth ratio may cause the flooding of the other overlapped meshes because the other meshes will not create a layer of interpolated cells to contain the flood.
+
+Increase the ``nearWallDist`` option in ADflow.
+    This option controls how compute cells are preserved near walls.
+    We usually use 0.01 for a full-scale aircraft mesh defined in meters.
+    Increasing ``nearWallDist`` will reduce the number of flood seeds.
+    Once you have a valid hole cutting, decrease ``nearwalldist`` to the minimum possible value.
+
+Check for sufficient overlap on the surface and in the volume.
+    The overlap should have at least 5 cells from each mesh.
+    Either extend the nearfield meshes or refine the background mesh until you have a 5 cell overlap in the off-wall direction.
+
+.. warning::
```

Comment:
This warning that I added

Reply:
Good catch, I forgot that this was not in there. I dont know how exactly flood seeds are tagged but I am guessing its either a bug or an edge case where the cell might be inside the geometry partially, but the cell center is outside of it. This warning is good as is.